threadpool_winxp.c

```diff
@@ -146,26 +141,25 @@ static uint32_t __stdcall threadpool_do_work(void *arg) {
 
     LOG_DEBUG("threadpool - worker 0x%" PRIx32 " - stopped\n", thread->id);
 
-    // Reduce thread count
-    threadpool->num_threads--;
     event_set(threadpool->lazy_cond);
     mutex_unlock(threadpool->queue_lock);
     return 0;
 }
 
 static void threadpool_create_thread_on_demand(threadpool_s *threadpool) {
     // Create new thread and add it to the list of threads
-    threadpool_thread_s *thread = (threadpool_thread_s *)calloc(1, sizeof(threadpool_thread_s));
-    if (!thread)
+    uint32_t id = 0;
+    HANDLE handle = (HANDLE)_beginthreadex(NULL, 0, threadpool_do_work, threadpool, 0, &id);
+    if (handle = -1)
```

Comment:
Should be `== -1`

Reply:
It's impossible, I remember I pressed `=` twice.